### PR TITLE
feat(topbar): add user dialog

### PR DESCRIPTION
**Pull request by nkrantz**
Add User Dialog component to the Topbar package

### CODE REVIEW

**Review by loreina — APPROVED** [review]

File: packages/paste-core/components/topbar/src/UserDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  children: PropTypes.node.isRequired,
+  'aria-label': PropTypes.string.isRequired,
+  element: PropTypes.string,
+};
+
+export const useUserDialogState = (
+  props: NonModalDialogPrimitivePopoverInitialState
+): NonModalDialogPrimitiveStateReturn => {
+  return useNonModalDialogPrimitiveState({...props, placement: 'bottom-start', gutter: 0});
+};
```
Comment (SiTaggart) [inline packages/paste-core/components/topbar/src/UserDialog.tsx:129]: suggestion: I'd potentially split some (if not all) of these out into their own files, under a UserDialog folder. Just so you can better understand what each thing is doing in isolation

File: packages/paste-core/components/topbar/src/UserDialogUserInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      </>
+    );
+  }
+);
+UserDialogUserInfo.displayName = 'UserDialogUserInfo';
+
+UserDialogUserInfo.propTypes = {
+  children: PropTypes.node.isRequired,
+  element: PropTypes.string,
+};
```
Comment (SiTaggart) [inline packages/paste-core/components/topbar/src/UserDialogUserInfo.tsx:100]: suggestion: same here, probably split all these into their own files.

**Review by SiTaggart — COMMENTED** [review]
Along with splitting things up into separate files, you need some customization stories and tests. Looking pretty sharp though

File: packages/paste-core/components/topbar/stories/userDialog.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    <UserDialogContainer name="User Name" icon={UserIcon} baseId="i-am-user-dialog">
+      <UserDialog aria-label="user menu" data-testid="basic-user-dialog">
+        <UserDialogUserInfo>
+          <UserDialogUserName>Name</UserDialogUserName>
+          <UserDialogUserEmail>email@email.com</UserDialogUserEmail>
+        </UserDialogUserInfo>
+      </UserDialog>
+    </UserDialogContainer>
+  );
+};
```
Comment (SiTaggart) [inline packages/paste-core/components/topbar/stories/userDialog.stories.tsx:34]: question: as we're extending the avatar do we need an image or `src` version to prove it works?

Comment (nkrantz) [reply packages/paste-core/components/topbar/stories/userDialog.stories.tsx:34]: yup good catch

**Review by SiTaggart — APPROVED** [review]

**Pull request merged**